assets/js/googlesitekit/components-gm2/SpinnerButton.js

```diff
@@ -31,6 +31,7 @@ export default function SpinnerButton( props ) {
 		className,
 		onClick = () => {},
 		isSaving = false,
+		spinnerOnLeft = false,
```

Comment:
Seeing as we've already diverged slightly from the IB, I'm wondering if we might want to go a step further still. Having an "on left" prop does feel _slightly_ clunky. What do you think about having a `spinnerPosition` prop instead, we could define and export a `SPINNER_POSITION` constant and the prop type could look like this (with the prop defaulting to `RIGHT`):

```js
spinnerPosition: PropTypes.oneOf( [ SPINNER_POSITION.LEFT, SPINNER_POSITION.RIGHT ] );
```

Or maybe `BEFORE` / `AFTER` to avoid explicit directional names.

We could also introduce both `--left` and `--right` suffixed CSS classes to avoid having to override for the left case (or before/after, again). Hopefully that all makes sense, please see what you think.

Reply:
That's all valid points and I've Updated the PR accordingly. I have one concern about making the API needlessly complicated, as we don't have types so it becoming a string prop with specific values instead of a simple boolean may take a little more getting used to.

Besides that, I took a look at the state of this component in RTL context, and realized that it's currently not ideal. In RTL, the spinner is positioned correctly, but the margin applied to it doesn't separate the spinner from the label, as it's applying a right margin. 

The prefer way to fix it is to use `margin-inline-start` and `margin-inline-end` which works on all modern browsers. But we can not use that because of SASS optimization, where:

````scss
.googlesitekit-plugin .mdc-button {

	&.googlesitekit-button-icon--spinner__running {

		&.googlesitekit-button-icon--spinner__after {

			.mdc-button__label {
				margin-inline-end: 8px;
			}
		}
	}
}
````

becomes 

````css
[dir="ltr"] .googlesitekit-plugin .mdc-button.googlesitekit-button-icon--spinner__running.googlesitekit-button-icon--spinner__after .mdc-button__label {
  margin-right: 8px;
}

[dir="rtl"] .googlesitekit-plugin .mdc-button.googlesitekit-button-icon--spinner__running.googlesitekit-button-icon--spinner__after .mdc-button__label {
  margin-left: 8px;
}
````

Which unfortunately doesn't apply anything when the html doesn't have a `dir` attribute, which is the default for WordPress for LTR languages. 

It would just work, if we could somehow disable this "optimization" for this rule, but I couldn't find any easy way to do it without upgrading sass to 3.1+!

As a fallback, I've added both direction sensitive and regular margins. This has no side effect on `LTR` layouts, but adds extra margin to the opposite direction on `RTL`, which while not ideal, is better than what we have today for `RTL`. 

Let me know what you think.

Reply:
Thanks @kuasha420.

> I have one concern about making the API needlessly complicated, as we don't have types so it becoming a string prop with specific values instead of a simple boolean may take a little more getting used to.

I know what you mean - I have to admit, this was playing on my mind a bit after I had made that suggestion. I decided to wait to see what you thought rather than muddying the water. There are definitely pros and cons for each approach. Seeing as you've implemented the `spinnerPosition` and it is arguably a bit semantically cleaner, although slightly more complicated with the addition of the string arg (it would be cleaner still if we had a nice TS type for it of course), let's keep this as it is now rather than spend more time on it. It's a pretty marginal point after all.

> It would just work, if we could somehow disable this "optimization" for this rule, but I couldn't find any easy way to do it without upgrading sass to 3.1+!
> 
> As a fallback, I've added both direction sensitive and regular margins. This has no side effect on `LTR` layouts, but adds extra margin to the opposite side on `RTL`, which while not ideal, is better than what we have today for RTL.

Nice work looking into this. I've a suggestion that might be preferable in terms of the end result, I'll post it on the diff rather than in this now outdated comment thread.